model/src/main/java/javasabr/mqtt/model/subscriber/tree/ConcurrentSubscriberTree.java

```diff
@@ -22,8 +21,8 @@ public ConcurrentSubscriberTree() {
   }
 
   @Nullable
-  public SingleSubscriber subscribe(MqttUser user, Subscription subscription) {
-    return rootNode.subscribe(0, user, subscription, subscription.topicFilter());
+  public SingleSubscriber subscribe(SingleSubscriber subscriber) {
```

Comment:
you break the abstraction

Reply:
I believe `SubscriptionService` has to create `SingleSubscriber`, not `SubscriberTree`